File: Dynamic Programming/Binomial Coefficient/BinomialCoefficient.py
class binom:
    def binomialCoef(self , n , k):
        C = [[0 for _ in range(k+1)] for _ in range(n+1)]
        for i in range(n+1):
            for j in range(min(i, k)+1):
                if (j == 0 or j == i):
                    C[i][j] = 1
                else:
                    C[i][j] = C[i-1][j-1] + C[i-1][j]

        return C[n][k]

if __name__ == "__main__":
    n = int(input("Enter the value of n: "))
    k = int(input("Enter the value of k: "))
    print("Value of nCk is " , binom().binomialCoef(n , k))

    

File: Dynamic Programming/Binomial Coefficient/BinomialCoefficient.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

int binomialCoeff(int n, int k){
	int C[n + 1][k + 1];
	for(int i = 0; i <= n; i++){
		for(int j = 0; j <= min(i, k); j++){
			if (j == 0 || j == i)
				C[i][j] = 1;
			else
				C[i][j] = C[i - 1][j - 1] + C[i - 1][j];
		}
	}

	return C[n][k];
}

int min(int a, int b){
    return (a < b) ? a : b;
}

int main(){
	int n , k;
    
    cout << "Enter the value of n: " << endl;
    cin >> n;
    cout << "Enter the value of k: " << endl;
    cin >> k;

	cout << "Value of nCk is " << binomialCoeff(n , k) << endl;
}

